bin/dist_get

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+
+die() {
+	echo "$@" >&2
+	exit 1
+}
+
+have_binary() {
+	type "$1" > /dev/null
+}
+
+check_writeable() {
+	printf "" > "$1" && rm "$1"
+}
+
+download() {
+	local url="$1"
+	local output="$2"
+
+	if [ -z "$url" ] || [ -z "$output" ]; then
```

Comment:
The usual way to test the number of arguments is to test "$#", like for example:

```
test "$#" -eq "1" || die "This program must be passed exactly 1 arguments" "Usage: $USAGE"
```